storage: improve TestPushTxnQueueCancel

tbg opened the pull request:
I was unable to reproduce

https://github.com/cockroachdb/cockroach/issues/19333

in >25k iterations, but I suspect an error was returned and the
test never got to check it. With these changes, the next time it
flakes we will have more intel.

Closes #19333.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/19410)
<!-- Reviewable:end -->

tamird commented on pkg/storage/push_txn_queue_test.go:
```diff
@@ -259,6 +259,11 @@ func TestPushTxnQueueCancel(t *testing.T) {
 	testutils.SucceedsSoon(t, func() error {
+		select {
+		case rwe := <-retCh:
+			t.Fatal("MaybeWaitForPush terminated prematurely: %+v", rwe)
```
Fatalf

tbg committed:
storage: improve TestPushTxnQueueCancel

I was unable to reproduce

https://github.com/cockroachdb/cockroach/issues/19333

in >25k iterations, but I suspect an error was returned and the
test never got to check it. With these changes, the next time it
flakes we will have more intel.

Closes #19333.

bdarnell commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/19410#-:-KwuuGbHL3QFy9yvYuCv:bnfp4nl)*
<!-- Sent from Reviewable.io -->
